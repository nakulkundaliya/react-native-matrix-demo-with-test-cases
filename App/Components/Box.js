import React, { Component } from 'react';
// import PropTypes from 'prop-types';
import { View, Text } from 'react-native';
import styles from './Styles/BoxStyle';

const Box = props => {
  const { value, nMatrix } = props;
  return (
    <View
      style={[
        styles.boxViewStyle,
        {
          height: 210 / nMatrix,
          width: 210 / nMatrix,
          backgroundColor: `rgb(255,200,${value})`
        }
      ]}
    >
      <Text>{value}</Text>
    </View>
  );
};

export default Box;
