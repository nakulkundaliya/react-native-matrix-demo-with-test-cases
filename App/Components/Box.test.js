import React from 'react';
import { shallow } from 'enzyme';
import Box from './Box';

describe('Box', () => {
  describe('Rendering', () => {
    it('should be defined', () => {
      expect(Box).toBeDefined();
    });
    it('should match to snapshot', () => {
      const component = shallow(<Box value={1} nMatrix={1} />);
      expect(component).toMatchSnapshot();
    });
  });
});
