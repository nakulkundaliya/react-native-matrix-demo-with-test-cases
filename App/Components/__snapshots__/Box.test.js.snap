// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Rendering should match to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Box
    nMatrix={1}
    value={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Text>
        1
      </Text>,
      "style": Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "#000",
          "borderWidth": 1,
          "justifyContent": "center",
          "marginRight": -1,
          "marginTop": -1,
        },
        Object {
          "backgroundColor": "rgb(255,200,1)",
          "height": 210,
          "width": 210,
        },
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": 1,
      },
      "ref": null,
      "rendered": 1,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Text>
          1
        </Text>,
        "style": Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderColor": "#000",
            "borderWidth": 1,
            "justifyContent": "center",
            "marginRight": -1,
            "marginTop": -1,
          },
          Object {
            "backgroundColor": "rgb(255,200,1)",
            "height": 210,
            "width": 210,
          },
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": 1,
        },
        "ref": null,
        "rendered": 1,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
